src/services/embeddings_service.py

```
@@ -0,0 +1,278 @@
+        for i in range(0, len(texts), batch_size):
+            batch = texts[i : i + batch_size]
+
+            try:
+                batch_embeddings = await self.create_embedding(batch, model)
+                all_embeddings.extend(batch_embeddings)
+
+                # Add small delay between batches to avoid rate limits
+                if i + batch_size < len(texts):
+                    await asyncio.sleep(0.1)
+
+            except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)